### DISCUSSION TITLE
Modifying data in running measurement (e.g. running average)

### DISCUSSION THREAD

**Discussion by paul-lehmann-unibas** [microsoft/Qcodes#5640] [Q&A] [answered]
Is it possible to modify data in a running measurement (except for adding more data ;) )? As far as I can tell the `DataSaver` only provides the `add_result` method.
My primary use case would be to save a running average of many repetitions of the same experiment. Since there are many repetitions, I do not want to save each one. It would be useful to update the running average in the database to protect against data loss in case of an error (although this could be done through a context manager) and to be able to plot the running average live from another process (e.g. plottr).

Thanks for any help and suggestions.

**Comment by jenshnielsen** [comment, marked as answer]
We don't really support modifying the data as we are sort of big on ensuring the provenance of the data so each value in a qcodes dataset corresponds to a specific measurement. I could imagine that you could do the averaging in memory and only write a snapshot of this to the database with some interval storing all the snapshots but by only doing it at certain intervals limiting the amount of data that you store.

**Reply by paul-lehmann-unibas** [reply]
I understand. Thanks for your input!
